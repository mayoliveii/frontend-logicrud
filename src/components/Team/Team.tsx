import { Flex } from "@chakra-ui/react";
import { Profile } from "./../Profile/Profile.tsx";

export function Team() {
  return (
    <Flex
      direction='row'
      alignItems='center'
      justifyContent='center'
      minHeight='80vh'
      gap='20px'
    >
      <Profile
        backgroundImage='https://tewbhv.vtexassets.com/arquivos/ids/158560/image-3fa4291f01d643978250dabb62fa556d.jpg?v=638064571894230000'
        profileImage='https://mayoliveii.s3.us-east-2.amazonaws.com/images/avatar-1.jpeg'
        name='Pedro Guilherme'
        role='Engenheiro de Dados'
        description='Pedro Ã© um engenheiro de dados que cria soluÃ§Ãµes para anÃ¡lise de grandes volumes de dados, 
        gerando insights importantes para os negÃ³cios.'
        tags={['PowerBI', 'SQL', 'Python']}
      />
      <Profile
        backgroundImage='https://www.codingem.com/wp-content/uploads/2021/10/juanjo-jaramillo-mZnx9429i94-unsplash-1024x683.jpg'
        profileImage='https://mayoliveii.s3.us-east-2.amazonaws.com/images/avatar-3.jpeg'
        name='Mayara Oliveira'
        role='Desenvolvedora de Software'
        description='Mayara, 24 anos, graduanda em AnÃ¡lise e Desenvolvimento de Sistemas, com mais de 2 anos 
        de experiÃªncia como desenvolvedora full stack. Futura colega de trabalho. ðŸ˜Š'
        tags={['Node', 'PostgreSQL', 'Vue']}
      />
      <Profile
        backgroundImage='https://p1p.com.br/blog/wp-content/uploads/2016/02/Dashboards-I.jpg'
        profileImage='https://mayoliveii.s3.us-east-2.amazonaws.com/images/avatar-2.jpg'
        name='John Doe'
        role='Product Manager'
        description='John Ã© um gerente de produto inovador, que lidera equipes multifuncionais para desenvolver 
        soluÃ§Ãµes de ponta que atendam Ã s necessidades do mercado e dos clientes.'
        tags={['Agile', 'JIRA', 'Scrum']}
      />
    </Flex>
  );
}