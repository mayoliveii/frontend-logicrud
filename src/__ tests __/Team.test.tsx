import { render } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { Team } from '../components/Team/Team';
import '@testing-library/jest-dom';

describe('Team', () => {
  it('renders Profile component and matches snapshot', () => {
    const { asFragment } = render(
      <ChakraProvider>
        <Team />
      </ChakraProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders Team component with Profile components', () => {
    const { getByText } = render(
      <ChakraProvider>
        <Team />
      </ChakraProvider>
    );

    expect(getByText('Pedro Guilherme')).toBeInTheDocument();
    expect(getByText('Engenheiro de Dados')).toBeInTheDocument();
    expect(getByText('Mayara Oliveira')).toBeInTheDocument();
    expect(getByText('Desenvolvedora de Software')).toBeInTheDocument();
    expect(getByText('John Doe')).toBeInTheDocument();
    expect(getByText('Product Manager')).toBeInTheDocument();
  });
});