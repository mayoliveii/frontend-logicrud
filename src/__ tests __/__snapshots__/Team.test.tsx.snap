// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team renders Profile component and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-44o3pl"
  >
    <div
      class="css-1mhvcxt"
    >
      <div
        class="css-j7qwjs"
      >
        <img
          class="chakra-image css-g1rp98"
          src="https://tewbhv.vtexassets.com/arquivos/ids/158560/image-3fa4291f01d643978250dabb62fa556d.jpg?v=638064571894230000"
        />
        <div
          class="css-9jmnaf"
        >
          <img
            class="chakra-image css-r3tnns"
            src="https://mayoliveii.s3.us-east-2.amazonaws.com/images/avatar-1.jpeg"
          />
          <p
            class="chakra-text css-13ufrgq"
          >
            Pedro Guilherme
          </p>
          <p
            class="chakra-text css-3jgsvj"
          >
            Engenheiro de Dados
          </p>
        </div>
        <p
          class="chakra-text css-mufhc5"
        >
          Pedro √© um engenheiro de dados que cria solu√ß√µes para an√°lise de grandes volumes de dados, 
        gerando insights importantes para os neg√≥cios.
        </p>
      </div>
      <div
        class="css-7m4n47"
      >
        <span
          class="chakra-badge css-aene3b"
        >
          PowerBI
        </span>
        <span
          class="chakra-badge css-aene3b"
        >
          SQL
        </span>
        <span
          class="chakra-badge css-aene3b"
        >
          Python
        </span>
      </div>
    </div>
    <div
      class="css-1mhvcxt"
    >
      <div
        class="css-j7qwjs"
      >
        <img
          class="chakra-image css-g1rp98"
          src="https://www.codingem.com/wp-content/uploads/2021/10/juanjo-jaramillo-mZnx9429i94-unsplash-1024x683.jpg"
        />
        <div
          class="css-9jmnaf"
        >
          <img
            class="chakra-image css-r3tnns"
            src="https://mayoliveii.s3.us-east-2.amazonaws.com/images/avatar-3.jpeg"
          />
          <p
            class="chakra-text css-13ufrgq"
          >
            Mayara Oliveira
          </p>
          <p
            class="chakra-text css-3jgsvj"
          >
            Desenvolvedora de Software
          </p>
        </div>
        <p
          class="chakra-text css-mufhc5"
        >
          Mayara, 24 anos, graduanda em An√°lise e Desenvolvimento de Sistemas, com mais de 2 anos 
        de experi√™ncia como desenvolvedora full stack. Futura colega de trabalho. üòä
        </p>
      </div>
      <div
        class="css-7m4n47"
      >
        <span
          class="chakra-badge css-aene3b"
        >
          Node
        </span>
        <span
          class="chakra-badge css-aene3b"
        >
          PostgreSQL
        </span>
        <span
          class="chakra-badge css-aene3b"
        >
          Vue
        </span>
      </div>
    </div>
    <div
      class="css-1mhvcxt"
    >
      <div
        class="css-j7qwjs"
      >
        <img
          class="chakra-image css-g1rp98"
          src="https://p1p.com.br/blog/wp-content/uploads/2016/02/Dashboards-I.jpg"
        />
        <div
          class="css-9jmnaf"
        >
          <img
            class="chakra-image css-r3tnns"
            src="https://mayoliveii.s3.us-east-2.amazonaws.com/images/avatar-2.jpg"
          />
          <p
            class="chakra-text css-13ufrgq"
          >
            John Doe
          </p>
          <p
            class="chakra-text css-3jgsvj"
          >
            Product Manager
          </p>
        </div>
        <p
          class="chakra-text css-mufhc5"
        >
          John √© um gerente de produto inovador, que lidera equipes multifuncionais para desenvolver 
        solu√ß√µes de ponta que atendam √†s necessidades do mercado e dos clientes.
        </p>
      </div>
      <div
        class="css-7m4n47"
      >
        <span
          class="chakra-badge css-aene3b"
        >
          Agile
        </span>
        <span
          class="chakra-badge css-aene3b"
        >
          JIRA
        </span>
        <span
          class="chakra-badge css-aene3b"
        >
          Scrum
        </span>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
